eelsdbconverter/__init__.py

```diff
@@ -49,60 +44,129 @@ def __init__(self):
+        sample.sample_title = file_data['title']
+
+        elements = file_data.get('elements')
         if elements is not None:
             if isinstance(elements, str):
                 elements = json.loads(elements)
-            material.atom_labels = elements
-        material.chemical_name = data.get('title')
-
-        experiment.m_create(
-            Method,
-            data_type='spectrum',
-            method_name='electron energy loss spectroscopy',
-            method_abbreviation='EELS')
-
-        section_data = experiment.m_create(Data)
-
-        archive.section_metadata.external_id = str(data.get('id'))
-        archive.section_metadata.external_db = 'EELSDB'
-
-        author = data.get('author')['name']
-        author = re.sub(r'\(.*\)', '', author).strip()
-        archive.section_metadata.coauthors = [Author(
-            first_name=' '.join(author.split(' ')[0:-1]),
-            last_name=author.split(' ')[-1])]
-        archive.section_metadata.comment = data.get('description')
-        archive.section_metadata.references = [
-            data.get('permalink'),
-            data.get('api_permalink'),
-            data.get('download_link')
-        ]
-
-        reference = data.get('reference')
-        if reference:
-            if isinstance(reference, str):
-                archive.section_metadata.references.append(reference)
-            elif isinstance(reference, dict):
-                if 'freetext' in reference:
-                    archive.section_metadata.references.append(
-                        re.sub(r'\r+\n+', '; ', reference['freetext']))
-                else:
-                    archive.section_metadata.references.append('; '.join([
-                        str(value).strip() for value in reference.values()
-                    ]))
-
-        section_data.repository_name = 'Electron Energy Loss Spectroscopy (EELS) database'
-        section_data.repository_url = 'https://eelsdb.eu'
-        section_data.entry_repository_url = data.get('permalink')
+            sample.elements = elements
+
+        # Experiment
+        experiment = metadata.m_create(Experiment)
+        experiment.method_name = 'electron energy loss spectroscopy'
+        experiment.method_abbreviation = 'EELS'
+        experiment.experiment_publish_time = datetime.strptime(
+            file_data.get('published'), '%Y-%m-%d %H:%M:%S')
+        edges = file_data.get('edges')
+        if edges is not None:
+            if isinstance(edges, str):
+                edges = json.loads(edges)
+            experiment.edges = edges
+
+        # Instrument
+        instrument = metadata.m_create(Instrument)
+        instrument.source_label = file_data['microscope']
+        device_settings = instrument.m_create(DeviceSettings)
+        device_settings.device_name = file_data['microscope']
+        device_settings.max_energy = file_data['max_energy']
+        if file_data.get('min_energy') is not None:
```

Comment:
I think `if 'min_energy' in file_data:` is easier to read (and might also be cheaper).

Reply:
I followed the coding style used in the previously written parser here. Do you want me to update it nevertheless? I can do it for all the definitions that I've done this way.